### PR TITLE
Admin menu: Add class for domain-only sites

### PR DESCRIPTION
**Pull request by mmtr**
Helps fixing https://github.com/Automattic/wp-calypso/issues/49128.
Required by D59042-code.

#### Changes proposed in this Pull Request:
Add a new admin menu class for domain-only sites which contains a single menu item to manage the domain.

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
See D59042-code.

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-domain-only-admin-menu.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	 *
+	 * Options:
+	 * false - Calypso (Default).
+	 * true  - wp-admin.
+	 *
+	 * @return bool
+	 */
+	public function should_link_to_wp_admin() {
+		return false;
+	}
```
Comment (cpapazoglou) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-domain-only-admin-menu.php:28]: Do we need this one?

Comment (mmtr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-domain-only-admin-menu.php:28]: It's an abstract method of the base class that needs to be implemented.

**Review by cpapazoglou — APPROVED** [review]
✅ Tested on a domain only site 
✅ Unit tests are passing

LGTM!

**Review by cpapazoglou — APPROVED** [review]
Tests are still passing!

**Review by jeherve — APPROVED** [review]

**Comment by mmtr** [comment]
Deployed to WP.com in r223207-wpcom.

**Pull request merged**